File: benchmarks/py/fib_iter.py
import sys


MOD = 1_000_000_007


def fib_iter(n: int) -> int:
  if n <= 1:
    return n
  a, b = 0, 1
  for _ in range(2, n + 1):
    a, b = b, (a + b) % MOD
  return b


if __name__ == "__main__":
  n = int(sys.argv[1]) if len(sys.argv) > 1 else 0
  print(fib_iter(n))

File: benchmarks/js/fib_iter.js
const MOD = 1000000007;

function fibIter(n) {
  if (n <= 1) return n;
  let a = 0;
  let b = 1;
  for (let i = 2; i <= n; i++) {
    const c = (a + b) % MOD;
    a = b;
    b = c;
  }
  return b;
}

const n = process.argv[2] ? parseInt(process.argv[2]) : 0;
console.log(fibIter(n));
